org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsi/settings/XSISchemaLocationSplit.java

```diff
@@ -0,0 +1,22 @@
+package org.eclipse.lemminx.extensions.xsi.settings;
+
+import org.eclipse.lemminx.settings.XMLFormattingOptions;
+
+public enum XSISchemaLocationSplit {
+
+	onNamespace, onLocation, none;
+
+	private static final String XSI_SCHEMA_LOCATION_SPLIT = "xsiSchemaLocationSplit";
+
+	public static XSISchemaLocationSplit getSplit(XMLFormattingOptions formattingSettings) {
+		return getSplit(formattingSettings.getString(XSI_SCHEMA_LOCATION_SPLIT));
+	}
+
+	public static XSISchemaLocationSplit getSplit(String value) {
+		try {
+			return XSISchemaLocationSplit.valueOf(value);

```

Comment:
check for null, no need to generate an expensive Exception if we know we're returning none anyway

Reply:
fixed